packages/clay-popover/src/index.tsx

```diff
@@ -114,20 +144,19 @@ const ClayPopover = React.forwardRef<HTMLDivElement, IProps>(
 			ref = popoverRef as React.Ref<HTMLDivElement>;
 		}
 
-		const show = externalShow ? externalShow : internalShow;
+		const show =
+			typeof externalShow === 'boolean' ? externalShow : internalShow;
```

Comment:
I don't like naming as internal* external*. Why not `initialShow`?

Reply:
Also, it looks like you're defining show is a boolean on IProps. why are you checking if it's a boolean or not?

Reply:
Oh, because it can be `undefined`?

Reply:
Yes, also if it is `false` it will toggle, in the end this should be replaced by default controlled and uncontrolled in PR #4786